### PR TITLE
[americastestkitchen] Fix metadata extraction for ATK episodes

### PR DESCRIPTION
**Pull request by bm1549**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Although the API does not appear to have changed between [episode 582](https://www.americastestkitchen.com/api/v6/episodes/582) and [episode 705](https://www.americastestkitchen.com/api/v6/episodes/705), the existing code fails to extract the metadata correctly for episode 705. This PR updates the metadata extraction to work well with the ATK api.

### CODE REVIEW

**Comment by bm1549** [comment]
FYI @remitamine this may be related to https://github.com/ytdl-org/youtube-dl/commit/17b01228f84713df850912b554675bbfcbc62fb3

File: youtube_dl/extractor/americastestkitchen.py [hunk 1]
```diff
@@ -60,7 +80,11 @@ def _real_extract(self, url):
             'url': 'https://player.zype.com/embed/%s.js?api_key=jZ9GUhRmxcPvX7M3SlfejB6Hle9jyHTdk2jVxG7wOHPLODgncEKVdPYBhuz9iWXQ' % video['zypeId'],
             'ie_key': 'Zype',
             'description': clean_html(video.get('description')),
+            'timestamp': parse_iso8601(video.get('publishDate')),
             'release_date': unified_strdate(video.get('publishDate')),
+            'upload_date': unified_strdate(video.get('publishDate')),
```
Comment (remitamine) [inline youtube_dl/extractor/americastestkitchen.py:85]: remove this line(will be calculated automatically from `timestamp`).

File: youtube_dl/extractor/americastestkitchen.py [hunk 2]
```diff
@@ -33,6 +34,25 @@ class AmericasTestKitchenIE(InfoExtractor):
         'params': {
             'skip_download': True,
         },
+    }, {
+        'url': 'https://www.americastestkitchen.com/episode/705-simple-chicken-dinner',
```
Comment (remitamine) [inline youtube_dl/extractor/americastestkitchen.py:38]: Add a comment on why this test has been added.

File: youtube_dl/extractor/americastestkitchen.py [hunk 3]
```diff
@@ -60,7 +80,11 @@ def _real_extract(self, url):
             'url': 'https://player.zype.com/embed/%s.js?api_key=jZ9GUhRmxcPvX7M3SlfejB6Hle9jyHTdk2jVxG7wOHPLODgncEKVdPYBhuz9iWXQ' % video['zypeId'],
             'ie_key': 'Zype',
             'description': clean_html(video.get('description')),
+            'timestamp': parse_iso8601(video.get('publishDate')),
             'release_date': unified_strdate(video.get('publishDate')),
+            'upload_date': unified_strdate(video.get('publishDate')),
+            'episode_number': episode.get('number'),
+            'season_number': episode.get('season'),
```
Comment (remitamine) [inline youtube_dl/extractor/americastestkitchen.py:87]: use `int_or_none`.

**Review by remitamine — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/americastestkitchen.py [hunk 4]
```diff
@@ -80,11 +83,10 @@ def _real_extract(self, url):
             'url': 'https://player.zype.com/embed/%s.js?api_key=jZ9GUhRmxcPvX7M3SlfejB6Hle9jyHTdk2jVxG7wOHPLODgncEKVdPYBhuz9iWXQ' % video['zypeId'],
             'ie_key': 'Zype',
             'description': clean_html(video.get('description')),
-            'timestamp': parse_iso8601(video.get('publishDate')),
+            'timestamp': unified_timestamp(video.get('publishDate')),
```
Comment (bm1549) [inline youtube_dl/extractor/americastestkitchen.py:86]: The timestamp was not a valid iso8601 timestamp. This fixes that issue and the `timestamp` in the second test case proves it

File: youtube_dl/extractor/americastestkitchen.py (hunk 3 above)
Comment (remitamine) [reply youtube_dl/extractor/americastestkitchen.py]: same for `episode_number`(both lines where highlighted).

Comment (bm1549) [reply youtube_dl/extractor/americastestkitchen.py]: My mistake. Just fixed this!

**Pull request merged**